File: backend/src/models/Product.js
const mongoose = require('mongoose');

const CATEGORIES = ['Shoes', 'Clothing', 'Accessories', 'Electronics', 'Sports', 'Other'];

const productSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  price: { type: Number, required: true },
  imageUrl: { type: String, required: true },
  company: { type: String, required: true },
  category: { type: String, enum: CATEGORIES, default: 'Other' },
  sizes: [Number],
  colors: [Number],
  description: { type: String, default: '' },
});

module.exports = mongoose.model('Product', productSchema);

File: lib/models/product.dart
import 'package:flutter/material.dart';

class Product {
  final String id;
  final String title;
  final double price;
  final String imageUrl;
  final String company;
  final List<int> sizes;
  final List<Color> colors;
  final String description;

  const Product({
    required this.id,
    required this.title,
    required this.price,
    required this.imageUrl,
    required this.company,
    required this.sizes,
    this.colors = const [],
    this.description = '',
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as String,
      title: json['title'] as String,
      price: (json['price'] as num).toDouble(),
      imageUrl: json['imageUrl'] as String,
      company: json['company'] as String,
      sizes: (json['sizes'] as List<dynamic>).map((e) => e as int).toList(),
      colors: (json['colors'] as List<dynamic>? ?? [])
          .map((e) => Color(e as int))
          .toList(),
      description: json['description'] as String? ?? '',
    );
  }
}
